feat(config): add set_extra_header method (#422)
algolia/algoliasearch-client-ruby: upgrade_guide.md

### `Client`

#### `set_extra_header`

> The `set_extra_header` method has been moved from the Client to the `Algolia::AlgoliaConfig` class. You have to define your extra headers on Client instantiation.

Reviewer: ```suggestion
The `set_extra_header` method has been moved to the `Algolia::AlgoliaConfig` class. You have to define your extra headers before you instantiate the `Client`.
```

```ruby
# Before
client.set_extra_header('admin', 'admin-key')

# After

> # `Algolia::Search::Config` inherits from `Algolia::AlgoliaConfig` 

Reviewer: ```suggestion
# Algolia::Search::Config inherits from Algolia::AlgoliaConfig
```
